modules/README.md

```diff
@@ -308,8 +326,11 @@ interface ISafeProtocolSignatureValidatorManager {
 
 ### Security considerations
 
-A malicious actor can change the routing of signature validation flow by changing the 4 bytes of signature selector.
-Under such cases the signature validation should fail and revert the transaction. To prove that signature validation fails due to changed routing, formal verification can be used. This would be tackled in future.
+A malicious actor can change the routing of signature validation flow by adding/removing the 4 bytes of signature selector. Under such cases the signature validation should fail and revert the transaction.
+
+Suppose, a malicious actor removes the 4 bytes of signature selector from a payload meant for domain specific validation flow, submits the modified payload for validation such that validator manager uses default signature validation flow. In such case, as the account expects `messageHash` to be generated by including domain separator defined by the signature validator manager, the account is expected to revert the call as hash used for signing is different from hash used for recovery.
+
+Similarly, if a malicious actor adds 4 bytes of signature selector to a payload meant for default signature validation flow, submits the modified payload for validation such that validator manager uses domain specific signature validation flow. Even if permitted by the registry and account has set a signature validator that matches signature validator's domain separator, the validation would only pass on approval of signature validator contract. So here, the flow is secure as long as the signature validator contract is not malicious.
```

Comment:
```suggestion
A malicious actor can change the routing of the signature validation flow by adding/removing the 4 bytes of the signature selector. Under such cases, the signature validation should fail and revert the transaction.

Suppose a malicious actor removes the 4 bytes of the signature selector from a payload meant for domain-specific validation flow and submits the modified payload for validation such that the validator manager uses default signature validation flow. In such a case, as the account expects `messageHash` to be generated by including the domain separator defined by the signature validator manager, the account is expected to revert the call as the hash used for signing differs from the hash used for recovery.

Similarly, if a malicious actor adds 4 bytes of the signature selector to a payload meant for default signature validation flow, submits the modified payload for validation such that the validator manager uses domain-specific signature validation flow. Even if permitted by the registry and the account has set a signature validator that matches the signature validator's domain separator, the validation would only pass on approval of the signature validator contract. So here, the flow is secure as long as the signature validator contract is not malicious.
```

Sorry, not trying to be picky but IMO for the specs we have to be perfect 💅

Reply:
And can't recommend https://grammarly.com/ enough (great use of the software budget)